drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -703,8 +703,8 @@ static void adsp_mm_save_context(void *storage_buffer)
 	location += sizeof(uint32_t);
 
 	z_xtensa_cache_flush(
-		storage_buffer,
-		(uint32_t)location - (uint32_t)storage_buffer);
+		(void *)storage_buffer,
```

Comment:
it's already `void *`... Actually `void __sparse_cache *()` and friends should have the `__sparse_cache` annotation with their pointer arguments because those functions should only be called for cache aliases, but currently they don't do that https://github.com/zephyrproject-rtos/zephyr/blob/b1615880320d39b89e1b7b467d6b9e6b67d322be/include/zephyr/arch/xtensa/cache.h#L24 so no type-casting is needed here